[docs] New page to migrate pickers from v5 to v6 (#6472)
mui/mui-x: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md

# Migration from v5 to v6

<p class="description">This guide describes the changes needed to migrate the Date & Time Pickers from v5 to v6.</p>

## Start using the alpha release

In `package.json`, change the version of the data grid package to `next`.

```diff
-"@mui/x-date-pickers": "latest",
+"@mui/x-date-pickers": "next",
```

Using `next` ensures that it will always use the latest v6 alpha release, but you can also use a fixed version, like `6.0.0-alpha.0`.

## Breaking changes

Since v6 is a major release, it contains some changes that affect the public API.
These changes were done for consistency, improve stability and make room for new features.
Below are described the steps you need to make to migrate from v5 to v6.

### Update the format of the `value` prop

> The `value` prop of the pickers now expects a parsed value.
> Until now, it was possible to provide any format that your date management library was able to parse.
> For instance, you could pass `value={new Date()}` when using `AdapterDayjs`.
> 
> This behavior brought a lot of confusion and was removed.
> The format expected by the `value` prop is now the same as for any other prop holding a date.
> Here is the syntax to initialize a date picker at the current date for each adapter:

Reviewer: I tied to rewrite it to clearly distinguish what is before and after.

I took inspiration from a random [migration guideline from Google](https://developer.chrome.com/docs/workbox/migration/migrate-from-v4/#default-precache-manifest-replacement-point) where I like the "Previously ....., In v5 ...." structure

```suggestion
Previously, it was possible to provide any format that your date management library was able to parse.
For instance, you could pass `value={new Date()}` when using `AdapterDayjs`. 
This behavior brought a lot of confusion.

In v6, the format expected by the `value` prop is the same as for any other prop holding a date.
Here is the syntax to initialize a date picker at the current date for each adapter:
```

```tsx
// Date-fns
<DatePicker value={new Date()} />;

// Dayjs
import dayjs from 'dayjs';
<DatePicker value={dayjs()} />;

// Moment
import moment from 'moment';
<DatePicker value={moment()} />;

// Luxon
import { DateTime } from 'luxon';
<DatePicker value={DateTime.now()} />;
```

### Rename the `LeftArrowButton` slot

The component slot `LeftArrowButton` has been renamed `PreviousIconButton` on all pickers:

```diff
<DatePicker
  components={{
-   LeftArrowButton: CustomButton,
+   PreviousIconButton: CustomButton,
  }}

  componentsProps={{
-   leftArrowButton: {},
+   previousIconButton: {},
  }}
/>
```

### Rename the `RightArrowButton` slot

The component slot `RightArrowButton` has been renamed `NextIconButton` on all pickers:

```diff
<DatePicker
  components={{
-   RightArrowButton: CustomButton,
+   NextIconButton: CustomButton,
  }}

  componentsProps={{
-   rightArrowButton: {},
+   nextIconButton: {},
  }}
/>
```

### Replace the `DialogProps` prop

> The `DialogProps` prop has been replaced by a `dialog` component props slot on all responsive and mobile pickers:

Reviewer 2: What do you think about removing `slot` word for these cases to avoid confusion between `component slot`? 🤔 
```suggestion
The `DialogProps` prop has been replaced by a `dialog` component props on all responsive and mobile pickers:
```

```diff
<DatePicker
-  DialogProps={{ backgroundColor: 'red' }}
+  componentsProps={{ dialog: { backgroundColor: 'red }}}
/>
```

### Replace the `PaperProps` prop

The `PaperProps` prop has been replaced by a `desktopPaper` component props slot on all responsive and desktop pickers:

```diff
<DatePicker
-  PaperProps={{ backgroundColor: 'red' }}
+  componentsProps={{ desktopPaper: { backgroundColor: 'red }}}
/>
```

### Replace the `PopperProps` prop

The `PopperProps` prop has been replaced by a `popper` component props slot on all responsive and desktop pickers:

```diff
<DatePicker
-  PopperProps={{ onClick: handleClick }}
+  componentsProps={{ popper: { onClick: handleClick }}}
/>
```

### Replace the `TransitionComponent` prop

The `TransitionComponent` prop has been replaced by a `desktopTransition` component slot on all responsive and desktop pickers:

```diff
<DatePicker
-  TransitionComponent={Fade}

> +  components={{ desktopTransition: Fade }}

Reviewer 2: ```suggestion
+  components={{ DesktopTransition: Fade }}
```

-  startText="From"
-  endText="To"
+  localeText={{
+    start: 'From',
+    end: 'To',
+  }}
/>
```

### Rename the `locale` prop on `LocalizationProvider`

The `locale` prop of the `LocalizationProvider` component have been renamed `adapterLcoale`:

```diff
<LocalizationProvider
  dateAdapter={AdapterDayjs}
-  locale="fr"
+  adapterLocale="fr"
>
  {children}
</LocalizationProvider
```

### Rename the view components

The view components allowing to pick a date or parts of a date without an input have been renamed to better fit their usage:

```diff
-<CalendarPicker {...props} />
+<DateCalendar  {...props} />
```

```diff
-<DayPicker {...props} />
+<DayCalendar  {...props} />
```

```diff
-<CalendarPickerSkeleton {...props} />
+<DayCalendarSkeleton  {...props} />
```

```diff
-<MonthPicker {...props} />
+<MonthCalendar  {...props} />
```

```diff
-<YearPicker {...props} />
+<YearCalendar  {...props} />
```

Component names in the theme have changed as well:

```diff
-MuiCalendarPicker: {
+MuiDateCalendar: {
```

```diff
-MuiDayPicker: {
+MuiDayCalendar: {
```

```diff
-MuiCalendarPickerSkeleton: {
+MuiDayCalendarSkeleton: {
```

```diff
-MuiMonthPicker: {
+MuiMonthCalendar: {

> ```
> 
> ```diff

Reviewer: ```suggestion

```

Could put all the examples in the same diff to be consistent with the last section

![image](https://user-images.githubusercontent.com/45398769/196369551-913ee4e6-b6e1-4a7a-a726-18a61a2ad160.png)
